### PR TITLE
feat: add CrossFlicking Preset

### PR DESCRIPTION
**Pull request by malangfox**
## Details
CrossFlicking supports flicking that can move in both horizontal and vertical directions.
You can easily create UIs with panels with multiple categories with this preset.

#### Documentation
https://malangfox.github.io/egjs-flicking/Presets

#### Examples
- [JavaScript (Group)](https://codepen.io/malangfox/pen/XWLNyqj)
- [JavaScript (Data attribute)](https://codepen.io/malangfox/pen/VwJbdrN)
- [React (Group)](https://codesandbox.io/p/sandbox/lively-bird-wy2f32?file=%2Fsrc%2FApp.js)
- [React (Data attribute)](https://codesandbox.io/p/sandbox/wonderful-silence-z2tc5k?file=%2Fsrc%2FApp.js%3A10%2C69)

#### Rules
- Moving the first panel up in the vertical Flicking changes to the previous category and changes the index in the horizontal Flicking.
- Moving the last panel down in the vertical Flicking changes to the next category and changes the index in the horizontal Flicking.

### CODE REVIEW

File: packages/react-flicking/src/react-flicking/CrossGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2015 NAVER Corp.
+ * egjs projects are licensed under the MIT license
+ */
+import React, { ReactNode } from "react";
+
+const CrossGroup = React.forwardRef((props: { children?: ReactNode }, ref: React.Ref<HTMLDivElement>) => <div ref={ref}>{props.children}</div>);
```
Comment (daybrush) [inline packages/react-flicking/src/react-flicking/CrossGroup.tsx:7]: It seems that CrossGroup should also have Element properties (className).

File: packages/react-flicking/src/react-flicking/CrossFlicking.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    this._bindEvents();
+    this._checkPlugins();
+
+    if (props.status) {
+      flicking.setStatus(props.status);
+    }
+  }
+
+  public render() {
+    const props = this.props;
```
Comment (daybrush) [inline packages/react-flicking/src/react-flicking/CrossFlicking.tsx:95]: It seems like you'll need to filter only the parts you need from Cross's props.

File: packages/react-flicking/src/react-flicking/CrossFlicking.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      </Viewport>
+    );
+  }
+
+  protected _bindEvents() {
+    super._bindEvents();
+
+    Object.keys(SIDE_EVENTS).forEach((eventKey: keyof typeof EVENTS) => {
+      const eventName = SIDE_EVENTS[eventKey];
+      this._bindEvent(eventName);
```
Comment (daybrush) [inline packages/react-flicking/src/react-flicking/CrossFlicking.tsx:129]: Must be paired with unmount.

File: packages/react-flicking/src/react-flicking/CrossFlicking.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  onSideHoldStart: (e: CrossFlickingEvent<HoldStartEvent>) => {},
+  onSideHoldEnd: (e: CrossFlickingEvent<HoldEndEvent>) => {},
+  onSideMoveStart: (e: CrossFlickingEvent<MoveStartEvent>) => {},
+  onSideMove: (e: CrossFlickingEvent<MoveEvent>) => {},
+  onSideMoveEnd: (e: CrossFlickingEvent<MoveEndEvent>) => {},
+  onSideWillChange: (e: CrossFlickingEvent<WillChangeEvent>) => {},
+  onSideChanged: (e: CrossFlickingEvent<ChangedEvent>) => {},
+};
+
+class CrossFlicking extends Flicking {
```
Comment (daybrush) [inline packages/react-flicking/src/react-flicking/CrossFlicking.tsx:51]: Is the method available?

Comment (malangfox) [reply packages/react-flicking/src/react-flicking/CrossFlicking.tsx:51]: CrossFlicking extends Flicking in react, and Flicking includes a `[@withFlickingMethods](https://github.com/naver/egjs-flicking/blob/master/src/cfc/withFlickingMethods.ts#L20)` decorator that allows to use existing methods.

File: packages/react-flicking/src/react-flicking/CrossGroup.tsx (hunk 1 above)
Comment (malangfox) [reply packages/react-flicking/src/react-flicking/CrossGroup.tsx]: Fixed to use properties of elements in CrossGroup.

File: packages/react-flicking/src/react-flicking/CrossFlicking.tsx (hunk 3 above)
Comment (malangfox) [reply packages/react-flicking/src/react-flicking/CrossFlicking.tsx:129]: Extended from react-flicking, react-crossflicking calls destroy on unmount to remove all event handlers.

File: packages/react-flicking/src/react-flicking/CrossFlicking.tsx (hunk 2 above)
Comment (malangfox) [reply packages/react-flicking/src/react-flicking/CrossFlicking.tsx:106]: Added logic to filter props before passing them to the CrossFlicking instance.

**Review by daybrush — APPROVED** [review]

**Pull request merged**